core/execution/future/market.go

```diff
@@ -954,6 +954,10 @@ func (m *Market) PostRestore(ctx context.Context) error {
 func (m *Market) OnTick(ctx context.Context, t time.Time) bool {
 	defer m.onTxProcessed()
 
+	if !m.matching.CheckBook() {
```

Comment:
Would it make more sense to do this on block-end before we commit the state?

Reply:
I'm not sure if we should keep this tbh, but if we do, yeah we could.